server/backend/cron/movies.go

```diff
@@ -0,0 +1,236 @@
+				continue
+			} else {
+				log.WithFields(logFields).Info("created movie")
+			}
+		}
+	}
+	return nil
+}
+
+type imdbResults struct {
+	Year       string
+	Runtime    string
+	Genre      string
+	Director   string
+	Actors     string
+	Plot       string
+	ImdbRating string `json:"imdbRating"`
+}
+
+func getIMDB(id string) (*imdbResults, error) {
+	url := "https://www.omdbapi.com/?r=json&v=1&i=" + id + "&apikey=" + os.Getenv("OMDB_API_KEY")
+	resp, err := http.Get(url)
+	if err != nil {
+		log.WithField("url", url).WithError(err).Error("Error while getting response for request")
+		return nil, err
+	}
+	// check if the api key is valid
+	if resp.StatusCode == http.StatusUnauthorized {
+		return nil, errors.New("missing or invalid api key for omdb (environment variable OMDB_API_KEY)")
+	}
+	// other errors
+	if resp.StatusCode != http.StatusOK {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			log.WithError(err).Warn("Unable to read http body")
+			return nil, err
+		} else {
+			log.WithField("status", resp.StatusCode).WithField("status", resp.Status).WithField("body", string(body)).Error("error while getting imdb movie")
+			return nil, errors.New("error while getting imdb movie")
+		}
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			log.WithField("url", url).WithError(err).Error("Error while closing body")
+		}
+	}(resp.Body)
+	// parse the response body
+	var res imdbResults
+	err = json.NewDecoder(resp.Body).Decode(&res)
+	if err != nil {
+		log.WithField("url", url).WithError(err).Error("Error while unmarshalling imdbResults")
+		return nil, err
+	}
+	return &res, nil
+}
+
+// extractTUFilmWebsite scrapes the imdb id and fullDescription from the tu-film website
+// url: url of the tu-film website, e.g. https://www.tu-film.de/programm/view/1204
+func extractTUFilmWebsite(url string) (string, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", errors.New("error while getting response for request")
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			log.WithError(err).Error("Error while closing body")
+		}
+	}(resp.Body)
+	// parse the response body
+	doc, err := goquery.NewDocumentFromReader(resp.Body)
+	if err != nil {
+		log.WithError(err).Error("Error while parsing document")
+		return "", err
+	}
+
+	// extract the imdb link
+	imdbLinks := doc.Find("a").FilterFunction(func(i int, s *goquery.Selection) bool {
+		href, hrefExists := s.Attr("href")
+		return hrefExists && strings.Contains(href, "imdb.com/title/")
+	})
+	if imdbLinks.Length() == 0 {
+		return "", errors.New("no imdb link found")
+	}
+	if imdbLinks.Length() > 1 {
+		log.WithField("url", url).Warn("more than one imdb link found. using first one")
+	}
+	// extrat the imdb id from the link
+	href, _ := imdbLinks.First().Attr("href")
+	href = strings.TrimSuffix(href, "/")
+	hrefParts := strings.Split(href, "/")
+	imdbID := hrefParts[len(hrefParts)-1]
```

Comment:
This is the perfect example for a unit test. If you put it into a separate function to separate it from the get operation, it shouldn't be too hard hopefully.